src/web/app/src/hooks/use-preferred-theme.ts

```diff
@@ -1,27 +1,26 @@
 import { useEffect } from 'react';
-import { useLocalStorage, useMedia } from 'react-use';
+import { useLocalStorage } from 'react-use';
 
 /**
  * Combination of localStorage for remembering the user's preference between
  * loads of the app, and initial logic to get the browser's preferred colour.
  */
 export default function usePreferredTheme() {
-  const isDarkThemePreferred = useMedia('(prefers-color-scheme: dark)', false);
-  const [preferredTheme, setPreferredTheme] = useLocalStorage(
+  const [preferredTheme, setPreferredTheme] = useLocalStorage<String>(
```

Comment:
I think the issue you're having with the build is related to this. Instead of type `String`, you can create a type that's something like:

```ts
type ThemeType = 'light-default' | 'light-high-contrast' | 'dark-default' | 'dark-dim';
```

Reply:
Thanks for great feedback. Let me try again.

Reply:
@manekenpix Sorry that I couldn't check build errors on my local. I have some spaces in my folder name which produces invalid folder error. I made some improvement in type checking and making const variables for repeated use. Please review and let me know if you have any feedback. Thanks